wlauto/devices/android/gem5/__init__.py

```diff
@@ -0,0 +1,820 @@
+            raise DeviceError('Attempting to execute "{}" as root on unrooted device.'.format(command))
+        if busybox:
+            if not self.is_rooted:
+                raise DeviceError('Attempting to execute "{}" with busybox. '.format(command) +
+                                  'Busybox can only be deployed to rooted devices.')
+            command = ' '.join([self.busybox, command])
+        if background:
+            self.logger.debug("Attempt to execute in background. Not supported "
+                              "in gem5, hence ignored.")
+        return self.gem5_shell(command, as_root=as_root)
+
+    def dump_logcat(self, outfile, filter_spec=None):
+        """ Extract logcat from simulation """
+        self.logger.info("Extracting logcat from the simulated system")
+        filename = outfile.split('/')[-1]
+        command = 'logcat -d > {}'.format(filename)
+        self.gem5_shell(command)
+        self.pull_file("{}".format(filename), outfile)
+
+    def clear_logcat(self):
+        """Clear (flush) logcat log."""
+        if self._logcat_poller:
+            return self._logcat_poller.clear_buffer()
+        else:
+            return self.gem5_shell('logcat -c')
+
+    def disable_selinux(self):
+        """ Disable SELinux. Overridden as parent implementation uses ADB """
+        api_level = int(self.gem5_shell('getprop ro.build.version.sdk').strip())
+
+        # SELinux was added in Android 4.3 (API level 18). Trying to
+        # 'getenforce' in earlier versions will produce an error.
+        if api_level >= 18:
+            se_status = self.execute('getenforce', as_root=True).strip()
+            if se_status == 'Enforcing':
+                self.execute('setenforce 0', as_root=True)
+
+    def get_properties(self, context):  # pylint: disable=R0801
+        """ Get the property files from the device """
+        for propfile in self.property_files:
+            try:
+                normname = propfile.lstrip(self.path.sep).replace(self.path.sep, '.')
+                outfile = os.path.join(context.host_working_directory, normname)
+                if self.is_file(propfile):
+                    self.execute('cat {} > {}'.format(propfile, normname))
+                    self.pull_file(normname, outfile)
+                elif self.is_directory(propfile):
+                    self.get_directory(context, propfile)
+                    continue
+                else:
+                    continue
+            except DeviceError:
+                # We pull these files "opportunistically", so if a pull fails
+                # (e.g. we don't have permissions to read the file), just note
+                # it quietly (not as an error/warning) and move on.
+                self.logger.debug('Could not pull property file "{}"'.format(propfile))
+
+        # This is duplicated from the AndroidDevice class, as we want to run
+        # this code without executing the BaseLinuxDevice implementation
```

Comment:
So, this is actually slightly modified from the original - I should update the comment. As we use the same terminal as is used for dmesg etc. for a gem5 system, I have changed the code to write  the output from 'dumpsys window' to a file and pull the file, as opposed to relying on the output from the shell command itself.

The way I see it, there are two choices. We keep the close-to duplicated code here for the gem5 device, or we do what you suggest above and move it to an internal method, but change it to redirect to a file and then pull the file. The latter would allow us to avoid code duplication, but does change how we process Android properties for all devices. Which do you prefer?